host.go

```diff
@@ -167,10 +161,38 @@ func (h *gpbftHost) SetAlarm(at time.Time) {
 // E.g. this might be: finalised tipset timestamp + epoch duration + stabilisation delay.
 func (h *gpbftHost) ReceiveDecision(decision *gpbft.Justification) time.Time {
 	h.log.Infof("got decision: %+v", decision)
-	//TODO propagate and save this for use in GetCanonicalChain
+	err := h.saveDecision(decision)
+	if err != nil {
+		h.log.Errorf("error while saving decision: %+v", err)
+	}
+
 	return time.Now().Add(2 * time.Second)
 }
 
+func (h *gpbftHost) saveDecision(decision *gpbft.Justification) error {
+	instance := decision.Vote.Instance
+	current, _, err := h.GetCommitteeForInstance(instance)
+	if err != nil {
+		return xerrors.Errorf("getting commitee for current instance %d: %w", instance, err)
+	}
+
+	next, _, err := h.GetCommitteeForInstance(instance + 1)
+	if err != nil {
+		return xerrors.Errorf("getting commitee for next instance %d: %w", instance+1, err)
+	}
+	powerDiff := certs.MakePowerTableDiff(current.Entries, next.Entries)
+
+	cert, err := certs.NewFinalityCertificate(powerDiff, decision)
+	if err != nil {
+		return xerrors.Errorf("forming certificate out of decision: %w", err)
+	}
+	err = h.client.certstore.Put(h.runningCtx, cert)
```

Comment:
IMO, we should spend the extra time and validate this certificate. We can either do that inside the certstore, or here.